bootloader/src/pyi_launch.c

```diff
@@ -355,6 +355,65 @@ pyi_launch_extract_binaries(ARCHIVE_STATUS *archive_status)
     return retcode;
 }
 
+/*
+ * Extract python exception message (string representation) from pvalue
+ * part of the error indicator data returned by PyErr_Fetch().
+ * Returns a copy of message string or NULL. Must be freed by caller.
+ */
+static char *
+_pyi_extract_exception_message(PyObject *pvalue)
+{
+    PyObject *pvalue_str;
+    const char *pvalue_cchar;
+    char *retval = NULL;
+
+    pvalue_str = PI_PyObject_Str(pvalue);
+    pvalue_cchar = PI_PyUnicode_AsUTF8(pvalue_str);
+    if (pvalue_cchar) {
+        retval = strdup(pvalue_cchar);
+    }
+    Py_DECREF(pvalue_str);
+
+    return retval;
+}
+
+/*
+ * Extract python exception traceback from error indicator data
+ * returned by PyErr_Fetch().
+ * Returns a copy of traceback string or NULL. Must be freed by caller.
+ */
+static char *
+_pyi_extract_exception_traceback(PyObject *ptype, PyObject *pvalue, PyObject *ptraceback)
+{
+    char *module_name;
+    PyObject *module;
+    char *retval = NULL;
+
+    /* Attempt to get a full traceback, source lines will only
+     * be available with --noarchive option */
+    module_name = "traceback";
+    module = PI_PyImport_ImportModule(module_name);
```

Comment:
Any reason for using an extra variable here?

Reply:
Fixed. Thanks!